File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int len = nums.length;

        int[] left = new int[len];
        int[] right = new int[len];

        //get the left sum
        int sum = 1;
        for(int i = 0; i < len; i++){
            left[i] = sum;
            sum *= nums[i];
            System.out.println(sum);
            
        }
        
        System.out.println("safsas");
        //get right
        sum = 1;
        for(int i = len - 1; i >= 0; i--){
            right[i] = sum; 
            sum *= nums[i];
        }

        for(int i = 0; i < len; i++){
            nums[i] = left[i] * right[i];
        }

        return nums;
    }
}  

File: Data Structures & Algorithms/three-integer-sum/submission-1.cs
public class Solution {
    public List<List<int>> ThreeSum(int[] nums) {
        /*
        store value in map and check to see if the value is present 
        to avoid dupes with ContainsValue
        */
        
        //two pointer 
        //sort list
        Array.Sort(nums);
        var list = new List<List<int>> ();

        for(int i = 0; i < nums.Length - 2; i++){
            //placeholder index
            //break since its impossible to get 0 if the array has been sorted
            if(nums[i] > 0) break;
            //move index along for unique value
            if (i > 0){
                if (nums[i]==nums[i-1]){
                    continue;
                }
            }
            //start left above placeholder {i,L,x,x,R}
            int left = i+1;
            int right = nums.Length - 1;
            while (left < right){
                //try find i + right + left that equals = 0
                //add to list
                int sum = nums[i] + nums[left] + nums[right];
                if (sum == 0){
                    list.Add(new List<int> {nums[i],nums[left],nums[right]});
                    left++;
                    right--;
                    while (left < right && nums[left] == nums[left-1]){
                        left++;
                    }
                }else if(sum < 0){
                    left++;
                }else{
                    right--;
                }
            }
        }

        return list; 


    }
}
